### PR TITLE
lint: add errchecks

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

Work towards enabling errcheck

ref #5059

### CODE REVIEW

**Review by erikgrinaker — COMMENTED** [review]
You're doing God's work here 👼

File: consensus/byzantine_test.go [hunk 1]
```diff
@@ -84,7 +84,8 @@ func TestByzantinePrevoteEquivocation(t *testing.T) {
 
 		eventBus := types.NewEventBus()
 		eventBus.SetLogger(log.TestingLogger().With("module", "events"))
-		eventBus.Start()
+		err = eventBus.Start()
+		assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply consensus/byzantine_test.go]: Looks like this was only updated in one file, it should be updated in all files.

[the remark this answers is not in the document]
Comment (marbar3778) [reply consensus/byzantine_test.go]: Ah okay, wasn't sure if that is what you meant. Will make the change

File: abci/client/socket_client.go [hunk 2]
```diff
@@ -500,5 +502,7 @@ func (cli *socketClient) stopForError(err error) {
 	cli.mtx.Unlock()
 
 	cli.Logger.Error(fmt.Sprintf("Stopping abci.socketClient for error: %v", err.Error()))
-	cli.Stop()
+	if err := cli.Stop(); err != nil {
+		cli.Logger.Error(fmt.Sprintf("error stopping abci.socketClient: %v", err.Error()))
```
Comment (melekes) [inline abci/client/socket_client.go:506]:
```suggestion
		cli.Logger.Error("Error stopping abci.socketClient", "err", err)
```

File: abci/tests/test_app/main.go [hunk 3]
```diff
@@ -57,17 +57,29 @@ func testCounter() {
 	cmd := exec.Command("bash", "-c", subCommand)
 	cmd.Stdout = os.Stdout
 	if err := cmd.Start(); err != nil {
-		log.Fatalf("starting %q err: %v", abciApp, err)
+		log.Fatalf("starting %q err: %v \n", abciApp, err)
```
Comment (melekes) [inline abci/tests/test_app/main.go:60]: ?

File: abci/tests/test_app/main.go [hunk 4]
```diff
@@ -57,17 +57,29 @@ func testCounter() {
...
 	}
-	defer cmd.Wait()
-	defer cmd.Process.Kill()
+	defer func() {
+		if err := cmd.Wait(); err != nil {
+			fmt.Printf("error while waiting for cmd to exit: %v \n", err)
```
Comment (melekes) [inline abci/tests/test_app/main.go:64]:
```suggestion
			log.Printf("error while waiting for cmd to exit: %v", err)
```

File: abci/tests/test_app/main.go [hunk 5]
```diff
@@ -57,17 +57,29 @@ func testCounter() {
...
+		}
+	}()
+	defer func() {
+		if err := cmd.Process.Kill(); err != nil {
+			fmt.Printf("error on process kill: %v \n", err)
```
Comment (melekes) [inline abci/tests/test_app/main.go:69]:
```suggestion
			log.Printf("error on process kill: %v", err)
```

File: abci/tests/test_app/main.go [hunk 6]
```diff
@@ -57,17 +57,29 @@ func testCounter() {
...
+		log.Fatalf("starting %q err: %v \n", abciApp, err)
 	}
-	defer cmd.Wait()
-	defer cmd.Process.Kill()
+	defer func() {
+		if err := cmd.Wait(); err != nil {
+			fmt.Printf("error while waiting for cmd to exit: %v \n", err)
+		}
+	}()
+	defer func() {
```
Comment (melekes) [inline abci/tests/test_app/main.go:67]: two defers can be combined

File: abci/tests/test_app/main.go [hunk 7]
```diff
@@ -57,17 +57,29 @@ func testCounter() {
...
 
 	if err := ensureABCIIsUp(abciType, maxABCIConnectTries); err != nil {
 		log.Fatalf("echo failed: %v", err)
 	}
 
 	client := startClient(abciType)
-	defer client.Stop()
+	defer func() {
+		if err := client.Stop(); err != nil {
+			fmt.Printf("error trying client stop: %v \n", err)
```
Comment (melekes) [inline abci/tests/test_app/main.go:80]:
```suggestion
			log.Printf("error trying client stop: %v", err)
```

File: cmd/tendermint/commands/reset_priv_validator.go [hunk 8]
```diff
@@ -58,7 +58,9 @@ func ResetAll(dbDir, addrBookFile, privValKeyFile, privValStateFile string, logg
 		logger.Error("Error removing all blockchain history", "dir", dbDir, "err", err)
 	}
 	// recreate the dbDir since the privVal state needs to live there
-	tmos.EnsureDir(dbDir, 0700)
+	if err := tmos.EnsureDir(dbDir, 0700); err != nil {
+		logger.Error("unable to recreate dbDir", "error", err)
```
Comment (melekes) [inline cmd/tendermint/commands/reset_priv_validator.go:62]:
```suggestion
		logger.Error("unable to recreate dbDir", "err", err)
```

File: consensus/reactor_test.go [hunk 9]
```diff
@@ -167,7 +171,8 @@ func TestReactorWithEvidence(t *testing.T) {
 
 		eventBus := types.NewEventBus()
 		eventBus.SetLogger(log.TestingLogger().With("module", "events"))
-		eventBus.Start()
+		err := eventBus.Start()
+		assert.NoError(t, err)
```
Comment (melekes) [inline consensus/reactor_test.go:175]:
```suggestion
		require.NoError(t, err)
```

**Review by erikgrinaker — APPROVED** [review]

**Pull request merged**